akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.actor.ActorSystem
+import akka.event.Logging
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.Http.ServerBinding
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * API MAY CHANGE - EXPERIMENTAL
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+// @akka.annotation.ApiMayChange // FIXME replace with real annotation once Akka dependency bumped
+trait HttpApp {
+
+  private val serverBinding = new AtomicReference[ServerBinding]()
+
+  /**
+   * Start a server on the specified host and port, using provided settings
+   * Note that this method is blocking
+   */
+  def startServer(host: String, port: Int, settings: ServerSettings): Unit = {
+    startServer(host, port, settings, None)
+  }
+
+  /**
+   * Start a server on the specified host and port, using provided settings and [[ActorSystem]]
+   * Note that this method is blocking
+   */
+  def startServer(host: String, port: Int, settings: ServerSettings, system: ActorSystem): Unit = {
+    startServer(host, port, settings, Some(system))
+  }
+
+  /**
+   * Start a server on the specified host and port, using provided settings and [[ActorSystem]] if present
+   * Note that this method is blocking
+   */
+  def startServer(host: String, port: Int, settings: ServerSettings, system: Option[ActorSystem]): Unit = {
+    implicit val theSystem = system.getOrElse(ActorSystem(Logging.simpleName(this).replaceAll("\\$", "")))
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = theSystem.dispatcher
+
+    val bindingFuture = Http().bindAndHandle(
+      handler = route,
+      interface = host,
+      port = port,
+      settings = settings)
+
+    bindingFuture.onComplete {
+      case Success(binding) ⇒
+        //setting the server binding for possible future uses in the client
+        serverBinding.set(binding)
+        postHttpBinding(binding)
+      case Failure(cause) ⇒
+        postHttpBindingFailure(cause)
+    }
+
+    Await.ready(
```

Comment:
Using ready instead of result because we want to stop waiting in case of any `Future` completing with a failure